test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go

```diff
@@ -0,0 +1,329 @@
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+)
+
+func TestAPIExportEndpointSlice(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	server := framework.SharedKcpServer(t)
+
+	// Create Organization and Workspaces
+	orgPath, _ := framework.NewOrganizationFixture(t, server)
+	var exportWs *tenancyv1alpha1.Workspace
+	exportClusterPath, exportWs := framework.NewWorkspaceFixture(t, server, orgPath, framework.WithName("export-workspace"))
+	t.Logf("Export Workspace %v", exportWs)
+	var partitionWs *tenancyv1alpha1.Workspace
+	partitionClusterPath, partitionWs := framework.NewWorkspaceFixture(t, server, orgPath, framework.WithName("partition"))
+	t.Logf("Partition Workspace %v", partitionWs)
+
+	cfg := server.BaseConfig(t)
+
+	var err error
+	kcpClusterClient, err := kcpclientset.NewForConfig(cfg)
+	require.NoError(t, err, "failed to construct kcp cluster client for server")
+
+	export := &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-export",
+		},
+	}
+
+	slice := &apisv1alpha1.APIExportEndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "my-slice",
+		},
+		Spec: apisv1alpha1.APIExportEndpointSliceSpec{
+			APIExport: apisv1alpha1.ExportBindingReference{
+				Path: exportClusterPath.String(),
+				Name: export.Name,
+			},
+		},
+	}
+
+	partition := &topologyv1alpha1.Partition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-partition",
+		},
+		Spec: topologyv1alpha1.PartitionSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"region": "apiexportendpointslice-test-region",
+				},
+			},
+		},
+	}
+
+	t.Logf("Creating an APIExportEndpointSlice with reference to a nonexistent APIExport")
+	sliceClient := kcpClusterClient.ApisV1alpha1().APIExportEndpointSlices()
+
+	_, err = sliceClient.Cluster(partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+	require.True(t, apierrors.IsForbidden(err), "no error creating APIExportEndpointSlice (admission should have declined it)")
+
+	t.Logf("Creating the missing APIExport")
+	exportClient := kcpClusterClient.ApisV1alpha1().APIExports()
+	_, err = exportClient.Cluster(exportClusterPath).Create(ctx, export, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating APIExport")
+
+	t.Logf("Retrying to create the APIExportEndpointSlice after the APIExport has been created")
+	require.Eventually(t, func() bool {
+		slice, err = sliceClient.Cluster(partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+		return err == nil
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected APIExportEndpointSlice creation to succeed")
+
+	require.Eventually(t, func() bool {
+		slice, err = kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsTrue(slice, apisv1alpha1.APIExportValid)
```

Comment:
should we also check any other conditions (if any) ? the test could ensure this was the only condition set.